src/net/ifaddrs/sockaddr.rs

```diff
@@ -0,0 +1,68 @@
+use std::net::{Ipv4Addr, Ipv6Addr, IpAddr};
+use std::mem::transmute;
+use libc;
+
+extern crate eui48;
+use self::eui48::{MacAddress, Eui48};
+
+/// Represents the actual data of an address in use by an interface.
+#[derive(Debug, Clone, PartialEq)]
+pub enum IfAddrValue {
+    IpAddr(IpAddr),
+    MacAddr(MacAddress),
+}
+
+impl From<IpAddr> for IfAddrValue {
+    fn from(ip: IpAddr) -> IfAddrValue {
+        IfAddrValue::IpAddr(ip)
+    }
+}
+
+impl From<MacAddress> for IfAddrValue {
+    fn from(mac: MacAddress) -> IfAddrValue {
+        IfAddrValue::MacAddr(mac)
+    }
+}
+
+/// Converts a `libc::sockaddr` into an `Option<IfAddrValue>`.
+///
+/// It returns `None` if the libc reports a type of address other than
```

Comment:
I've been chewing on this, and I've concluded that we're doing it wrong.  There are two problems:
1) Nix consumers can't extend this function to handle weird stuff like `AF_IPX`.
2) Handling link addresses is too platform specific.  That functionality probably doesn't belong in Nix at all.

Both of these problems share a common solution.  For unknown address families, `sockaddr_to_ifaddrvalue` should return `Other(&libc::sockaddr)` instead of `None`.  That way Nix consumers can do whatever they need to handle `AF_LINK` or anything else.

Reply:
I like that solution a lot, actually. I'll go ahead and implement that.